app/src/main/java/org/wikipedia/page/references/ReferenceDialog.kt

```diff
@@ -91,10 +93,25 @@ class ReferenceDialog : ExtendedBottomSheetDialogFragment() {
             binding.referenceText.movementMethod = LinkMovementMethodExt(callback()?.linkHandler)
         }
 
-        fun bindItem(idText: CharSequence?, contents: CharSequence?) {
+        fun bindItem(idText: CharSequence?, reference: PageReferences.Reference) {
             binding.referenceId.text = idText
             binding.root.post {
                 if (isAdded) {
+                    val contents = StringUtil.fromHtml(StringUtil.removeCiteMarkup(StringUtil.removeStyleTags(reference.html)))
+                    if (contents.isNullOrEmpty()) {
+                        // Inspect html for external links
+                        val links = StringUtil.extractHrefLinks(reference.html)
+                        links.add(links[0])
+                        var tags = ""
+                        for (i in 0 until links.size) {
+                            tags = tags.plus("<a href='https:${links[i]}'>[${i + 1}]</a>")
+                        }
+                        binding.referenceText.text = StringUtil.fromHtml(tags)
+                        binding.referenceText.movementMethod = LinkMovementMethod()
```

Comment:
The `referenceText` already has a movementMethod. Why do we need to give it a new one?